src/test/java/io/jenkins/plugins/analysis/warnings/StepsITest.java

```diff
@@ -51,6 +52,104 @@
  */
 @SuppressWarnings({"PMD.ExcessiveImports", "checkstyle:ClassDataAbstractionCoupling", "checkstyle:ClassFanOutComplexity"})
 public class StepsITest extends IntegrationTestWithJenkinsPerTest {
+
+    private static final String NO_QUALITY_GATE = "";
+
+    /**
+     * Runs a pipeline and verifies the {@code scanForIssues} step has some whitelisted methods.
+     */
+    @Test
+    public void shouldWhitelistScannerApi() {
+        WorkflowJob job = createPipelineWithWorkspaceFiles("recorder/checkstyle1.xml", "recorder/checkstyle2.xml");
+
+        configureScanner(job, "checkstyle1");
+        Run<?, ?> baseline = buildSuccessfully(job);
+
+        assertThat(getConsoleLog(baseline)).contains("[total=" + 3 + "]");
+        assertThat(getConsoleLog(baseline)).contains("[id=checkstyle]");
+
+        System.out.println("----- Console Log -----");
+        System.out.println(getConsoleLog(baseline));
+
+        configureScanner(job, "checkstyle2");
+        Run<?, ?> build = buildSuccessfully(job);
+        assertThat(getConsoleLog(build)).contains("[total=" + 4 + "]");
+        assertThat(getConsoleLog(build)).contains("[id=checkstyle]");
+    }
+
+    private void configureScanner(final WorkflowJob job, final String fileName) {
+        job.setDefinition(new CpsFlowDefinition("node {\n"
+                + "  stage ('Integration Test') {\n"
+                + "         def report = scanForIssues tool: checkStyle(pattern: '**/" + fileName + "*')\n"
+                + "         echo '[total=' + report.size() + ']' \n"
+                + "         echo '[id=' + report.getId() + ']' \n"
+                + "  }\n"
+                + "}", true));
+    }
+
+    /**
+     * Runs a pipeline and verifies the {@code publishIssues} step has whitelisted methods.
+     */
+    @Test
+    public void shouldWhitelistPublisherApi() {
+        WorkflowJob job = createPipelineWithWorkspaceFiles("recorder/checkstyle1.xml", "recorder/checkstyle2.xml");
+
+        configurePublisher(job, "checkstyle1", NO_QUALITY_GATE);
+        Run<?, ?> baseline = buildSuccessfully(job);
+
+        verifyApiResults(baseline, 3, 0, 0, "INACTIVE");
+
+        System.out.println("----- Console Log -----");
+        System.out.println(getConsoleLog(baseline));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [System.out.println is used](https://app.codacy.com/manual/uhafner/warnings-ng-plugin/pullRequest?prid=4771623)